File: backend/PropMate.Api/Models/Favourite.cs
using System.Text.Json.Serialization;

namespace PropMate.Api.Models;

public class Favourite
{
    public int Id { get; set; }
    public int PropertyId { get; set; }
    public int UserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation property
    [JsonIgnore]
    public Property? Property { get; set; }
}

File: mobile/propmate_mobile/lib/models/favourite.dart
class Favourite {
  final int id;
  final int propertyId;
  final int userId;

  Favourite({required this.id, required this.propertyId, required this.userId});

  factory Favourite.fromJson(Map<String, dynamic> json) {
    return Favourite(
      id: json['id'],
      propertyId: json['propertyId'],
      userId: json['userId'],
    );
  }
}
